### PR TITLE
[MISC] rename include/seqan3/io/{alignment => sam}_file 

### PR DESCRIPTION
**Pull request by marehr**
Part of many small rename PRs, tackles https://github.com/seqan/product_backlog/issues/290.

### CODE REVIEW

**Review by SGSSGene — CHANGES_REQUESTED** [review]
Can you sort the includes?

**Comment by marehr** [comment]
> Can you sort the includes?

Done, your script was handy :)

File: include/seqan3/io/all.hpp [hunk 1]
```diff
@@ -100,7 +100,7 @@
 
 #pragma once
 
-#include <seqan3/io/alignment_file/all.hpp>
+#include <seqan3/io/sam_file/all.hpp>
```
Comment (SGSSGene) [inline include/seqan3/io/all.hpp:103]: Falsche reinfolge... (hat mein skript das übersehen?)

**Review by SGSSGene — CHANGES_REQUESTED** [review]
One ordering error left. Other wise LGTM :+1:

File: include/seqan3/io/all.hpp [hunk 2]
```diff
@@ -100,7 +100,7 @@
...
 #include <seqan3/io/exception.hpp>
 #include <seqan3/io/record.hpp>
```
Comment (SGSSGene) [inline include/seqan3/io/all.hpp:105]: Falsche Reihenfolge:

```suggestion
#include <seqan3/io/exception.hpp>
#include <seqan3/io/record.hpp>
#include <seqan3/io/sam_file/all.hpp>
```

**Review by SGSSGene — APPROVED** [review]
:+1:

**Review by eseiler — APPROVED** [review]

**Pull request merged**